src/fides/api/ops/schemas/user_permission.py

```diff
@@ -1,23 +1,23 @@
-from typing import List
+from typing import List, Optional, Set
 
 from fastapi import HTTPException
 from pydantic import validator
 from starlette.status import HTTP_422_UNPROCESSABLE_ENTITY
 
 from fides.api.ops.api.v1.scope_registry import SCOPE_REGISTRY
-from fides.lib.oauth.schemas.user_permission import (
-    UserPermissionsCreate as UserPermissionsCreateLib,
-)
-from fides.lib.oauth.schemas.user_permission import (
-    UserPermissionsEdit as UserPermissionsEditLib,
-)
-from fides.lib.oauth.schemas.user_permission import (
-    UserPermissionsResponse as UserPermissionsResponseLib,
-)
+from fides.api.ops.schemas.base_class import BaseSchema
+from fides.lib.oauth.roles import ROLE_REGISTRY
 
 
-class UserPermissionsCreate(UserPermissionsCreateLib):
-    """Data required to create a FidesUserPermissions record"""
+class UserPermissionsCreate(BaseSchema):
+    """Data required to create a FidesUserPermissions record
+
+    Users will generally be assigned role(s) directly which are associated with many scopes,
+    but we also will continue to support the ability to be assigned specific individual scopes.
+    """
+
+    scopes: List[str] = []
```

Comment:
this is a nit, but since we know beforehand all of the defined scopes/roles this could be an Enum

Reply:
Yes good point, @allisonking was [also looking into making scopes an enum](https://github.com/ethyca/fides/pull/2682/files), so I'd need to coordinate there, and could also make the roles enums.  I think I was just patterning this way because scopes were already a list of strings, but an enum makes more sense.

Reply:
OK I've added a role enum here https://github.com/ethyca/fides/pull/2671/files#diff-2a6e5b613afe739bdfbefc63a01ce84e08aa97a31a25f5e3eaafe41affb8ff10R40 and will leave the scopes enum for Allison -